### PR TITLE
Add a documentation page on MPI apps

### PR DESCRIPTION
**Pull request by WardLT**
# Description

Adds a documentation page which explains how to launch MPI apps from Parsl.

See #2723 

## Type of change

Choose which options apply, and delete the ones which do not apply.

- Documentation update

### CODE REVIEW

File: docs/userguide/mpi_apps.rst [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+The :class:`~parsl.executors.high_throughput.executor.HighThroughputExecutor` (HTEx) is the default execution provider
+available through Parsl.
+Parsl Apps which invoke MPI code require a dedicated HTEx configured such that every Parsl app
+will have access to any of the nodes available within a block,
+and Apps that will invoke the MPI launcher with the correct settings.
+
+Configuring the Provider
+++++++++++++++++++++++++
+
+Parsl must be configured to deploy exactly one executor per execution block.
```
Comment (benclifford) [inline docs/userguide/mpi_apps.rst:29]: when you say "executor" here I think maybe you mean something more like "htex worker"?

File: docs/userguide/apps.rst [hunk 2]
```diff
@@ -191,3 +191,11 @@ Limitations
...
 
 1. Environment variables are not supported.
+
+
+MPI Apps
+--------
+
+Applications which employ MPI to span multiple nodes are a special case of Bash or Python apps,
+and require special modification of Parsl's `execution environment <execution.html>`_ to function.
+We describe support for MPI applications `in a later section <mpi_apps.html>`_.
```
Comment (benclifford) [inline docs/userguide/apps.rst:201]: dont' use academic style "we" in user guides

File: docs/userguide/mpi_apps.rst (hunk 1 above)
Comment (benclifford) [reply docs/userguide/mpi_apps.rst:29]: or htex worker pool?

File: docs/userguide/mpi_apps.rst [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+This part is simple.
+Instead of defining `a launcher <execution.html#launchers>`_ which will
+place an executor on each node in the block, simply use the :class:`~parsl.launchers.SimpleLauncher`.
+
+It is also necessary to specify the desired number of blocks for the executor.
+Parsl cannot determine the number of blocks needed to run a set of MPI Apps,
```
Comment (benclifford) [inline docs/userguide/mpi_apps.rst:35]: does this document ever assume you'll be running a single MPI-app-instance (phrasing?) across nodes provisioned from several blocks? i think not?

File: docs/userguide/mpi_apps.rst [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+so they must bet set explicitly (see `Issue #1647 <https://github.com/Parsl/parsl/issues/1647>`_).
+The easiest route is to set the ``max_blocks`` and ``min_blocks`` of the provider
+to the desired number of blocks.
+
+Configuring the Executor
+++++++++++++++++++++++++
+
+Configure the executor to launch a number of workers equal to the number of MPI tasks per block.
+First set the ``max_workers`` to the number of MPI Apps you will run per block,
+then set ``cores_per_worker=1e-6`` to avoid limiting by cores available to the executor.
```
Comment (benclifford) [inline docs/userguide/mpi_apps.rst:45]: i'm not really clear in my head what setting cores_per_worker is going to do here... you've already said that the head node of the block will only run max_workers simultaneous apps.  what does cores_per_worker then do beyond that?

**Review by benclifford — APPROVED** [review]

**Comment by benclifford** [comment]
merging this now because it's definitely good enough, with some comments for later consideration in the review

File: docs/userguide/mpi_apps.rst (hunk 3 above)
Comment (WardLT) [reply docs/userguide/mpi_apps.rst:35]: No, it does not. I'll add a note about blocks being a hard boundary for MPI applications to the "Limitations"

File: docs/userguide/mpi_apps.rst (hunk 4 above)
Comment (WardLT) [reply docs/userguide/mpi_apps.rst:45]: Setting cores_per_worker fixes this

**Pull request merged**